### DISCUSSION TITLE
TextArea Height and Rows

### DISCUSSION THREAD

**Discussion by gkassym** [twilio-labs/paste#2180] [General] [answered]
Hi there 

Why can’t we pass the height of the textarea as a prop? Why is Rows hard-coded?

Thanks

<img width="663" alt="Screenshot 2022-02-02 at 10 17 32 PM" src="https://user-images.githubusercontent.com/95677319/152236288-4e9d3fa1-c4f1-40eb-8f26-b2ff029209d9.png">

**Comment by gloriliale** [comment, marked as answer]
Hi @gkassym! We've touched on this in [a recent thread](https://github.com/twilio-labs/paste/discussions/2099), and there's [an older discussion thread](https://github.com/twilio-labs/paste/discussions/700) that might also be useful context, but the simplest answer is that this is intentional, and that the limit is there for the component to be consistent and visually different from text inputs.

Let us know if you're working on a particular use case so that we are able to help if needed!

**Reply by askel4dd** [reply]
For context, this is what we are trying to build, Chat input basically and it make sense for it to be 1 row in height in a defualt state. I read through the discussions, I guess we will build something ourselves for now 


<img width="562" alt="Screenshot 2022-02-03 at 13 26 28" src="https://user-images.githubusercontent.com/7442658/152334436-774b18e3-498b-4e58-9efb-864dcac695f6.png">

<img width="561" alt="Screenshot 2022-02-03 at 13 26 39" src="https://user-images.githubusercontent.com/7442658/152334434-c3fc698d-c5a3-4276-9f40-ba12ffe5c163.png">
